subsys/net/lib/sockets/sockets_select.c

```diff
@@ -197,3 +200,75 @@ int zsock_select(int nfds, zsock_fd_set *readfds, zsock_fd_set *writefds,
 
 	return num_selects;
 }
+
+#ifdef CONFIG_USERSPACE
+static int z_vrfy_zsock_select(int nfds, zsock_fd_set *readfds,
+			       zsock_fd_set *writefds,
+			       zsock_fd_set *exceptfds,
+			       struct zsock_timeval *timeout)
+{
+	zsock_fd_set *readfds_copy, *writefds_copy, *exceptfds_copy;
+	size_t fds_size;
+	int ret;
+
+	/* Copy fds array from user mode */
+	if (size_mul_overflow(nfds, sizeof(zsock_fd_set), &fds_size)) {
```

Comment:
I don't think I understand the purpose of this multiplication. Unless I miss something, the `readfds`/`writefds`/`exceptfds` should always point to a single instance of `zsock_fd_set` structure, regardless of the `nfds` parameter. `nfds` just speicifes the range of the fds to verify in the `zsock_fd_set` bitfield.

Reply:
I copied this from poll implementation, but that was not a proper thing to do. I will change this.